### PR TITLE
[iOS] - Fix #27841: Added API for use_legacy_component and callback to the AdblockService

### PR DESCRIPTION
**Pull request by Brandon-T**
Resolves https://github.com/brave/brave-browser/issues/27841

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by cuba — APPROVED** [review]

File: components/brave_shields/browser/ad_block_component_installer.cc [hunk 1]
```diff
@@ -60,6 +60,17 @@ const char kAdBlockIosDefaultDatComponentBase64PublicKey[] =
     "5HcH/heRrB4MvrE1J76WF3fvZ03aHVcnlLtQeiNNOZ7VbBDXdie8Nomf/QswbBGa"
     "VwIDAQAB";
 
+const char kAdBlockIosComponentName[] = "Brave Ad Block Updater";
+const char kAdBlockIosComponentId[] = "iodkpdagapdfkphljnddpjlldadblomo";
+const char kAdBlockIosComponentBase64PublicKey[] =
```
Comment (kylehickinson) [inline components/brave_shields/browser/ad_block_component_installer.cc:65]: Can we make it more obvious that this component isn't iOS specific?

Comment (kylehickinson) [reply components/brave_shields/browser/ad_block_component_installer.cc:65]: Either that or find a way to share the component defined in the [ad_block_service.cc](https://github.com/brave/brave-core/blob/7963b1cb3df30d08bc1bb8144fe422f6b1f16fb6/components/brave_shields/browser/ad_block_service.cc#L44)

**Comment by kylehickinson** [comment]
Based on this solution keeping the properties for install location & filter lists are we going to only call it once on brave-ios side? (cc @cuba)

**Comment by kjozwiak** [comment]
Removed the `1.49.x` milestone as this hasn't landed in `master` yet.

**Comment by cuba** [comment]
> Based on this solution keeping the properties for install location & filter lists are we going to only call it once on brave-ios side? (cc @cuba)

If I understand you correctly, yes we will call it only once. Technically we already only attach the callback once so the code won't change much on the iOS side. As for the filter lists, those are only retrieved once but there shouldn't be any reason why we can't access it multiple times.

**Review by cuba — CHANGES_REQUESTED** [review]

File: components/brave_shields/browser/ad_block_component_installer.cc (hunk 1 above)
Comment (cuba) [reply components/brave_shields/browser/ad_block_component_installer.cc:65]: This is all temporary and can be removed once we switch to the non-legacy version

**Comment by cuba** [comment]
@Brandon-T its been some time and I have forgotten what is missing to merge this.

**Comment by cuba** [comment]
Since this PR is really out of date and things have changed a lot since its creation I have created a new PR here which uses newer components: 
https://github.com/brave/brave-core/pull/20137

**Pull request closed without merging**